### ISSUE TITLE
`tmt test list` generates wrong `when` condition

### ISSUE THREAD

**Issue by bachradsusi** [psss/tmt#495]
~~~
$ git clone https://src.fedoraproject.org/tests/selinux.git
$ cd selinux
$ tmt test lint -f libselinux/validatetrans 
$ git diff 
index 886443b102a7..de9516a9f772 100644
--- a/libselinux/validatetrans/main.fmf
+++ b/libselinux/validatetrans/main.fmf
@@ -1,4 +1,5 @@
 path: /libselinux/validatetrans
-relevancy:
-- "distro < fedora-32: False"
-
+adjust:
+-   enabled: false
+    when: distro ~< fedora-32
+    continue: false
~~~

According to  https://fmf.readthedocs.io/en/latest/context.html#comparison  `when: distro ~< fedora-32` can't be true.

**Comment by lukaszachy** [comment]
@psss: This week we  decided to limit the precision based on right (rule) side (for Context.matches) Should that apply for this case as well?

**Comment by bachradsusi** [comment]
Note that the repo moved forward. If you want to reproduce it, use `git checkout 5be146b0853f`

**Comment by psss** [comment]
I'd say that when `fedora` is included in the condition `tmt test list --fix` should just not use minor distro comparison as it does not make any sense there.

**Comment by bachradsusi** [comment]
It's not about *fedora*, it does the same wrong conversion for rhel as well, see https://src.fedoraproject.org/tests/selinux/pull-request/148#_1__5

~~~
--- a/kernel/labeled-cephfs/main.fmf
+++ b/kernel/labeled-cephfs/main.fmf
@@ -4,6 +4,8 @@ summary: Test for correct CephFS SELinux labeling
 description: |
     This TC verifies that CephFS security label support works as expected.
 duration: 30m
-relevancy:
-    - "distro < rhel-8: False"
 tier: 2
+adjust:
+-   enabled: false
+    when: distro ~< rhel-8
+    continue: false
~~~

**Comment by psss** [comment]
Ah, I see. So I guess the special `~` operators should be used in the coversion from the old format only when the distro on the right side contains a minor version. I don't think this should/could be handled in `fmf` as we should still be able to do for example both `component < bash-5` and `component < bash-5.2` which should work as a "regular" version comparison. @lukaszachy, agreed?

**Comment by psss** [comment]
Should be fixed by #496.
